### PR TITLE
Send events

### PR DESCRIPTION
**Pull request by steve-dickinson**
https://eaflood.atlassian.net/browse/SFI-1582

### CODE REVIEW

File: app/config/index.js [hunk 1]
```diff
@@ -54,6 +54,8 @@ value.qcSubscription = mqConfig.qcSubscription
 value.manualTopic = mqConfig.manualTopic
 value.qcManualTopic = mqConfig.qcManualTopic
 value.qcManualSubscription = mqConfig.qcManualSubscription
+value.eventTopic = mqConfig.eventTopic
 value.dbConfig = dbConfig
+value.eventTopic = mqConfig.eventTopic
```
Comment (johnwatson484) [inline app/config/index.js:59]: Duplicate with line 57

File: app/messaging/update-requests-awaiting-debt-data.js [hunk 2]
```diff
@@ -27,6 +28,7 @@ const prepareForReprocessing = async (paymentRequest, debtType, recoveryDate) =>
     where: { paymentRequestId: paymentRequest.paymentRequestId }
   })
   await removeHold(paymentRequest.schemeId, paymentRequest.frn)
+  await sendProcessingRouteEvent(paymentRequest, 'debt', 'response')
```
Comment (johnwatson484) [inline app/messaging/update-requests-awaiting-debt-data.js:31]: Maybe things like `debt` and `response` should be constants?

**Review by johnwatson484 — COMMENTED** [review]
README and async API needs updating

File: app/event/send-processing-ack-event.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const raiseEvent = require('./raise-event')
+const { v4: uuidv4 } = require('uuid')
+
+const sendProcessingAckEvent = async (message, action) => {
+  const event = {
+    id: message?.correlationId ?? uuidv4(),
+    name: `payment-request-${action}-event`,
```
Comment (johnwatson484) [inline app/event/send-processing-ack-event.js:7]: Don't think these all need `event` on the end

File: app/config/index.js (hunk 1 above)
Comment (simonnineteen90) [reply app/config/index.js:59]: removed

File: app/event/raise-event.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const { PublishEvent } = require('ffc-pay-event-publisher')
+const config = require('../config')
+
+const raiseEvent = async (event, status = 'success', error = '') => {
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply app/event/raise-event.js:4]: removed

File: app/event/send-processing-return-event.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  const invoiceNumber = message.invoiceNumber
+  const frn = message.frn
+
+  const { correlationId, paymentRequestNumber, agreementNumber } = await getPaymentSchemeByInvoiceAndFrn(invoiceNumber, frn)
+
+  const event = {
+    id: correlationId ?? uuidv4(),
+    name: 'payment-request-return-event',
+    type: 'return',
+    message: 'Payment request return from DAX',
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply app/event/send-processing-return-event.js:15]: updated

File: app/event/send-processing-ack-event.js (hunk 3 above)
Comment (simonnineteen90) [reply app/event/send-processing-ack-event.js:7]: removed from all

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**